main.go

```diff
@@ -87,9 +105,19 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Cache: cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
```

Comment:
The reason I never went for this change is that I wasn't sure of the implication of it as this would be a partial cache. More resources apart from PipelineRuns are cached (eg. Releases) and I'm not sure if this would cause only PipelineRuns to be cached or it's just a filter. It was easier to understand in previous versions of controller-runtime for sure...

Probably this is fine though and this is just a selector prior caching PipelineRuns.

Reply:
Nice one. I found this in the docs:
```
IMPORTANT NOTE Requests to a client backed by a filtered cache for objects that do not match the filter will never return anything. In other words, filtered caches make the filtered-out objects invisible to the client.
```

IIUC we need to add all other objects we want to cache here as well. I think the PipelineRun is the only that needs an actual filter, while the other ones should be here, but (somehow) without a filter. I will check how to do it.

Reply:
Anyhow, I am waiting to see how e2e will behave with this change.

Reply:
We should implement filters for Releases (not now but in the future). You will also need to list ReleasePlans and ReleasePlanAdmissions and probably ReleaseServiceConfigs, Applications and Snapshots as I think those get cached as well. Not sure about the latter. Resources involved in controllers are cached for sure.

Reply:
Apparently, only objects that are watched by the controllers needs to be added to the manager cache, but I will test both cases (with all objects we need vs all objects we watch) and see how it behaves.